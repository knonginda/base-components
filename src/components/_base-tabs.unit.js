import BaseTabs from './_base-tabs'

const mountTabs = (options = {}) => {
  return mount(BaseTabs, {
    ...createComponentMocks({
      slots: {
        default: `
          <BaseTab title="Tab Title1" active>Tab Content1</BaseTab>
          <BaseTab title="Tab Title2">Tab Content2</BaseTab>
          <BaseTab title="Tab Title3" disabled>Tab Content3</BaseTab>
        `,
      },
    }),
    ...options,
  })
}

describe('@components/_base-tabs', () => {
  let wrapper = mountTabs()

  it('should open the first tab', () => {
    expect(
      wrapper
        .findAll('.tabContent > div')
        .at(0)
        .attributes().style
    ).not.toBe('display: none;')

    expect(
      wrapper
        .findAll('.tabContent > div')
        .at(1)
        .attributes().style
    ).toBe('display: none;')

    expect(
      wrapper
        .findAll('.tabContent > div')
        .at(2)
        .attributes().style
    ).toBe('display: none;')

    expect(
      wrapper
        .findAll('.tabItem')
        .at(0)
        .classes()
    ).toContain('active')

    expect(
      wrapper
        .findAll('.tabItem')
        .at(1)
        .classes()
    ).not.toContain('active')

    expect(
      wrapper
        .findAll('.tabItem')
        .at(2)
        .classes()
    ).not.toContain('active')
  })

  it('should open the second tab', () => {
    wrapper
      .findAll('.tabItem')
      .at(1)
      .trigger('click')
    expect(
      wrapper
        .findAll('.tabContent > div')
        .at(0)
        .attributes().style
    ).toBe('display: none;')

    expect(
      wrapper
        .findAll('.tabContent > div')
        .at(1)
        .attributes().style
    ).not.toBe('display: none;')

    expect(
      wrapper
        .findAll('.tabContent > div')
        .at(2)
        .attributes().style
    ).toBe('display: none;')

    expect(
      wrapper
        .findAll('.tabItem')
        .at(0)
        .classes()
    ).not.toContain('active')

    expect(
      wrapper
        .findAll('.tabItem')
        .at(1)
        .classes()
    ).toContain('active')

    expect(
      wrapper
        .findAll('.tabItem')
        .at(2)
        .classes()
    ).not.toContain('active')
  })

  it('should be medium size', () => {
    expect(wrapper.element.outerHTML).toContain('medium')
  })

  it('should disabeld third tab', () => {
    expect(
      wrapper
        .findAll('.tabItem')
        .at(2)
        .classes()
    ).toContain('disabled')
  })
})
