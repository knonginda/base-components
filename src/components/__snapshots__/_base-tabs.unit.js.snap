// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/_base-tabs should render its content when size is large 1`] = `
<div class="large tab">
  <ul>
    <li class="active tabItem">Tab Title1</li>
    <li class="tabItem">Tab Title2</li>
    <li class="disabled tabItem">Tab Title3</li>
  </ul>
  <div class="tabContent">
    <div class="">Tab Content1</div>
    <div class="" style="display: none;">Tab Content2</div>
    <div class="disabled" style="display: none;">Tab Content3</div>
  </div>
</div>
`;

exports[`@components/_base-tabs should render its content when size is small 1`] = `
<div class="small tab">
  <ul>
    <li class="active tabItem">Tab Title1</li>
    <li class="tabItem">Tab Title2</li>
    <li class="disabled tabItem">Tab Title3</li>
  </ul>
  <div class="tabContent">
    <div class="">Tab Content1</div>
    <div class="" style="display: none;">Tab Content2</div>
    <div class="disabled" style="display: none;">Tab Content3</div>
  </div>
</div>
`;
